src/hotspot/share/gc/shenandoah/mode/shenandoahPassiveMode.cpp

```diff
@@ -26,6 +26,7 @@
 #include "gc/shenandoah/heuristics/shenandoahHeuristics.hpp"
 #include "gc/shenandoah/heuristics/shenandoahPassiveHeuristics.hpp"
 #include "gc/shenandoah/mode/shenandoahPassiveMode.hpp"
+#include "gc/shenandoah/shenandoahGeneration.hpp"
```

Comment:
Are we sure we need this include?  I see no other changes in this source file so it looks suspicious.

Reply:
We do need that include because `initialize_heuristics` takes a `ShenandoahGeneration*` argument. I added the include here to fix the zero build. I'll modify the interface on the modes to use `ShenandoahHeapCharacteristics` instead. The code for initializing heuristics is a bit over-engineered and I might be able to simplify it further.